kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/writer/DhcpServerConfigWriterTest.java

```diff
@@ -347,14 +424,32 @@ private void thenKuraExceptionIsCaught() {
         assertTrue(this.isKuraException);
     }
 
+    /*
+     * Utilities
+     */
+
     @After
-    public void clean() {
-        File dhcpConfigFile = new File("/tmp/dhcpd-eth0.conf");
-        if (dhcpConfigFile.exists()) {
-            dhcpConfigFile.delete();
+    public void cleanUp() throws IOException {
+        dhcpServerMock.close();
+        this.mockFiles.delete();
+        this.mockFiles.create();
+    }
+
+    private class PropertiesBuilder {
```

Comment:
This next comment ties together with https://github.com/eclipse/kura/pull/4520/files#r1156204051.

Again: I really like the builder pattern but... is it really necessary here? Couldn't a `givenNetworkPropertiesWith(String key, Object value)` method suffice and obtain the overall same results without the need of an additional class?

Keep in mind that this is how it was already implemented in all other tests using `NetworkProperties`. See:

https://github.com/eclipse/kura/blob/b01a8133d4d34ee2533b7efa0bd82825bfa2a89e/kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java#L96-103

https://github.com/eclipse/kura/blob/b01a8133d4d34ee2533b7efa0bd82825bfa2a89e/kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NetworkPropertiesTest.java#L72-74

Reply:
Yes it might be more intuitive this way. I wanted to try something new but it seems to be not the right choice.